src/libmongoc/tests/test-mongoc-read-prefs.c

```diff
@@ -805,29 +806,32 @@ test_read_prefs_mongos_max_staleness (void)
    prefs = mongoc_read_prefs_new (MONGOC_READ_SECONDARY_PREFERRED);
    mongoc_read_prefs_set_max_staleness_seconds (prefs, 120);
 
-   /* exhaust cursor is required so the driver downgrades the OP_QUERY find
-    * command to an OP_QUERY legacy find */
```

Comment:
Since the new behavior only affects 5.1+ servers, can this test change be reverted?

The mock server in this test reports wire version 5 (corresponding to 3.4 server version).

Reply:
Will revert.